docs/src/content/docs/guides/stages.mdx

```diff
@@ -0,0 +1,111 @@
+---
+title: "Call Stages"
+description: Use Ultravox API to create dynamic, multi-stage conversations with granular system prompts.
+---
+
+import { Steps, Code } from '@astrojs/starlight/components';
+
+The Ultravox API's Call Stages functionality allows you to create dynamic, multi-stage conversations by providing granular system prompts as your agent progresses through the interaction. Stages enable more complex and nuanced agent interactions, giving you fine-grained control over the conversation flow.
+
+## Understanding Call Stages
+
+Call Stages ("Stages") provide a way to segment a conversation into distinct phases, each with its own system prompt and potentially different parameters. This enables interactions that can adapt and change focus as the conversation progresses.
+
+Key points to understand about Stages:
+
+**Dynamic System Prompts** → Stages allow you to give granular system prompts to the model as the conversation progresses.
+
+**Flexibility** → You have full control to determine when and how you want the conversation to progress to the next stage.
+
+**Thoughtful Design** → Implementing stages requires careful planning and consideration of the conversation structure. Consider how to handle stage transitions and test thoroughly to ensure a natural flow to the conversation.
+
+**Maintain Context** → Think about how the agent will maintain context about the user between stages if you need to ensure a coherent conversation.
+
+:::note[Advanced Feature]
+Stages require planning and careful implementation. Stages are likely not required for simple use cases.
+:::
+
+## Creating and Managing Stages
+
+To implement Call Stages in your Ultravox application, follow these steps:
+
+#### 1. Plan Your Stages
+Determine the different phases of your conversation and what prompts or parameters should change at each stage.
+
+#### 2. Implement a Stage Change Tool
+Create a custom tool that will trigger stage changes when called. A new call stage is created when your tool responds with a header called `X-Ultravox-Response-Type` set to `new-stage` in the response.
+
+Unless overridden, stages inherit all properties of the existing call. Here is the list of all call properties that can and cannot be changed during a new stage:
+
+| property            | change with new stage? |
+|-------------------- | ---------------------- |
+| systemPrompt        | Yes                    |
+| temperature         | Yes                    |
+| voice               | Yes                    |
+| languageHint        | Yes                    |
+| initialMessages     | Yes                    |
+| selectedTools       | Yes                    |
+| firstSpeaker        | Yes                    |
```

Comment:
firstSpeaker can't change either - it only matters for the first stage after all.

Reply:
updated